src/components/Text.js

```diff
@@ -1,39 +1,45 @@
 // @flow
 import React from "react";
 import { Text as RnText, StyleSheet } from "react-native";
-import Markdown from "react-native-easy-markdown";
-
 import type { TextProps } from "react-native/Libraries/Text/TextProps";
-
+import Markdown from "react-native-easy-markdown";
 import { blackColor, lightNavyBlueColor } from "../constants/colors";
 
 export type TextType = "h1" | "h2" | "h3" | "h4" | "text" | "small" | "price";
+export type TextColor = "lightNavyBlueColor";
 
 type Props = {
-  type?: TextType,
-  color?: "lightNavyBlueColor",
-  markdown?: boolean,
+  type: TextType,
+  color: ?TextColor,
+  markdown: boolean,
```

Comment:
Is this not an optional prop anymore? We've still got a default for it below.

Or has the way we define these changed in the new flow?

Reply:
Well, this is a flow special. I definitely need to find the docs for this:

Normally you define an optional property with the question mark (`markdown?: boolean`). When used in a react component, flow then requires you to define a default prop for this. When you use the prop in the component it will always be defined, since the default value is used. However flow still thinks the value can be undefined.

Interestingly, only for component classes, there is another way. As soon as you define a default prop for any prop, it automatically becomes optional for flow. It does not need to be marked as optional using the question mark. Then flow knows that the prop will not be undefined when you use it. Only in this case the default value for `color` is `undefined`. That's why I put the question mark in front of the type.

As I said, I did not read this in any documentation. I actually found this by accident. So this might not be entirely correct.